### ISSUE TITLE
discrepency in evaluation of the validation set 

### ISSUE THREAD

**Issue by naarkhoo** [catboost/catboost#1577]
Thank you for developing such a powerful package.
I am using x_val and y_val to validate the model 

`model.fit(train_pool, eval_set=(x_val, y_val), verbose=True)`

the result looks like 

```
148:	learn: 0.8267842	test: 0.8113082	best: 0.8113082 (148)	total: 1m 45s	remaining: 711ms
149:	learn: 0.8268854	test: 0.8112834	best: 0.8113082 (148)	total: 1m 46s	remaining: 0us

bestTest = 0.8113081945
bestIteration = 148

Shrink model to first 149 iterations.
```

then I use the model and predict `x_val`
```

y_pred_cat = model.predict_proba(x_val, )
y_pred_cat_prob = [item[1] for item in y_pred_cat]
y_pred_cat_lab = [item[1] > 0.5 for item in y_pred_cat]
str('Accuracy :' +  str(metrics.accuracy_score(y_val, y_pred_cat_lab)))
```

but then I get 

`'Accuracy :0.6864370200384435'`

I expected to get sth very close to 81% ! I understand the model is over fit bu then I wonder what is the use of those evaluation of the validation set ! am I missing something ?

**Comment by LyzhinIvan** [comment]
Hi, @naarkhoo!
Are you sure that you set eval_metric='Accuracy' in the model? Do you have categorical features? Please, provide all options of your model.

**Comment by naarkhoo** [comment]
@LyzhinIvan - 

```
model = ctb.CatBoostClassifier(
    iterations=50,
    learning_rate=0.1,
    use_best_model=True,
    depth=10,
    reg_lambda=0.1,
    loss_function='Logloss',
    eval_metric='Accuracy',
    scale_pos_weight=10/6,
    max_bin=10,
    verbose=200
)

Also, yes I do have 25 number of categorical features together with 68 number of numeric features. my data is slightly imbalanced 60% positive - so I used `scale_pos_weight`. without using this parameter the accuracy is 73%. I noticed if I increase the scale_pos_weight I can get better performance shown during the training (even with the validation set around 80%). but if I use the same validation it won't be that high and in fact around 73% ! so I guess 'scale_pos_weight' makes it look good during iteration but not afterwards. 

I appreciate if you interpret this behavior. Thanks

```

**Comment by LyzhinIvan** [comment]
> ```
>     scale_pos_weight=10/6,
> ```

`scale_pos_weight=10/6` equals to `class_weights=[1, 10/6]` and all metrics calculated during the training use this weights by default. This means that 0-label objects are treated with weight=1 and 1-label objects are treated with weight=10/6. But when you call `metrics.accuracy_score`, you don't specify any weights now, so all objects are treated with weight=1.
To get similar results you can do one of these things:
1) Disable weights for Accuracy (`eval_metric='Accuracy:use_weights=False'`)
2) Pass `sample_weight` to `metrics.accuracy_score`.
